mozjs/js/rust/magic_codegen/src/lib.rs

```diff
@@ -35,18 +37,32 @@ fn match_struct(ast: &syn::DeriveInput, variant: &syn::VariantData) -> quote::To
         .split('_')
         .next()
         .expect("Should have a '_' in the magic dom struct name");
+    let dom_flag = if name.starts_with("DOM") {
+        quote::Ident::from("jsapi::JSCLASS_IS_DOMJSCLASS")
```

Comment:
There are lots of types that are DOM classes that won't start with "DOM": `Node`, `Console`, etc.

Did you find it necessary to have this flag set? AFAICT, this is only used outside of the JS engine by embedders (gecko/servo) and I think we want to make whether it is set or not orthogonal to the magic dom stuff.